crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs

```diff
@@ -33,9 +33,55 @@ pub struct KnownSenderData {
     pub device_id: Option<OwnedDeviceId>,
 
     /// The cross-signing key of the user who established this session.
+    #[serde(
+        serialize_with = "serialize_ed25519_key",
+        deserialize_with = "deserialize_sender_msk_base64_or_array"
+    )]
     pub master_key: Box<Ed25519PublicKey>,
 }
 
+/// In an initial version the master key was serialized as an array of number,
+/// it is now exported in base64. This code adds backward compatibility.
+pub(crate) fn deserialize_sender_msk_base64_or_array<'de, D>(
```

Comment:
I am not familiar with this serde visitor pattern. Let me know if it is sane or if there is a prefferred method? I am not sure about the error mappings